lib/curly/presenter.rb

```diff
@@ -52,6 +52,10 @@ def initialize(context, options = {})
           end
         end
 
+        if value.respond_to?(:call)
+          value = value.call
```

Comment:
I'm not sure I understand the question. (Also, I'd still prefer the block to be eval'ed on the presenter instance.)